moveit_planners/ompl/ompl_interface/src/detail/ompl_constraints.cpp

```diff
@@ -0,0 +1,318 @@
+{
+}
+
+void PositionConstraint::parseConstraintMsg(const moveit_msgs::Constraints& constraints)
+{
+  ROS_INFO_STREAM_NAMED(LOGNAME, "Parsing position constraint for OMPL constrained state space.");
+  bounds_.clear();
+  bounds_ = positionConstraintMsgToBoundVector(constraints.position_constraints.at(0));
+  ROS_INFO_STREAM_NAMED(LOGNAME, "Parsed x constraints" << bounds_[0]);
+  ROS_INFO_STREAM_NAMED(LOGNAME, "Parsed y constraints" << bounds_[1]);
+  ROS_INFO_STREAM_NAMED(LOGNAME, "Parsed z constraints" << bounds_[2]);
+
+  // extract target position and orientation
+  geometry_msgs::Point position =
+      constraints.position_constraints.at(0).constraint_region.primitive_poses.at(0).position;
+  target_position_ << position.x, position.y, position.z;
+  tf::quaternionMsgToEigen(constraints.position_constraints.at(0).constraint_region.primitive_poses.at(0).orientation,
+                           target_orientation_);
+
+  link_name_ = constraints.position_constraints.at(0).link_name;
+  ROS_INFO_STREAM_NAMED(LOGNAME, "Position constraints applied to link: " << link_name_);
+}
+
+Eigen::VectorXd PositionConstraint::calcError(const Eigen::Ref<const Eigen::VectorXd>& x) const
+{
+  return target_orientation_.matrix().transpose() * (forwardKinematics(x).translation() - target_position_);
+}
+
+Eigen::MatrixXd PositionConstraint::calcErrorJacobian(const Eigen::Ref<const Eigen::VectorXd>& x) const
+{
+  return target_orientation_.matrix().transpose() * robotGeometricJacobian(x).topRows(3);
+}
+
+/******************************************
+ * Equality constraints
+ * ****************************************/
+EqualityPositionConstraint::EqualityPositionConstraint(const robot_model::RobotModelConstPtr& robot_model,
+                                                       const std::string& group, const unsigned int num_dofs)
+  : BaseConstraint(robot_model, group, num_dofs)
+{
+}
+
+void EqualityPositionConstraint::parseConstraintMsg(const moveit_msgs::Constraints& constraints)
+{
+  ROS_INFO_STREAM_NAMED(LOGNAME, "Parsing equality position constraint for OMPL constrained state space.");
+  bounds_.clear();
+
+  std::vector<double> dims = constraints.position_constraints.at(0).constraint_region.primitives.at(0).dimensions;
+
+  is_dim_constrained_ = { false, false, false };
+  for (std::size_t i{ 0 }; i < dims.size(); ++i)
+  {
+    if (dims[i] < equality_constraint_threshold_)
+    {
+      if (dims[i] < getTolerance())
+      {
+        ROS_ERROR_NAMED(
+            LOGNAME,
+            "Dimension %i of position constraint is smaller than the tolerance used to evaluate the constraints. "
+            "This will make all states invalid and planning will fail :( Please use a value between %f and %f. ",
+            i, getTolerance(), equality_constraint_threshold_);
+      }
+      is_dim_constrained_.at(i) = true;
+    }
+  }
+
+  ROS_INFO_STREAM_NAMED(LOGNAME, "X dimension constraint? " << is_dim_constrained_[0]);
```

Comment:
This should probably be prettier and DEBUG level

Reply:
done